core/platform/cloud_translate/cloud_translate_emulator.py

```diff
@@ -0,0 +1,73 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An emulator that mocks the core.platform.cloud_translate API. This emulator
```

Comment:
Is this only for the backend test env, or is it for the local dev environment too? Please clarify in the docstring.

More generally, where is this file actually used? I don't see anything else referencing it in this PR.

Reply:
I added some clarification, the emulator is not currently used, rather, it exists for backend testing and possibly local dev environment.
In the local dev environment, it's much easier to create a free tier GCP and set up authentication (I added a link and instructions for doing so). But in the case that authentication is not an option, the emulator can be used for local testing.

Reply:
Sorry, I still don't understand. In platform/models.py, when calling import_cloud_translate_services(), we only return the non-emulator module. If we want the local dev environment to work, it needs to work "out of the box", just like all our other third-party platform dependencies. 

You'll probably want the code in platform/models.py to choose the correct implementation based on whether we're in EMULATOR_MODE or not, similar to how import_taskqueue_services works. I suggest changing the naming here to be consistent with that example, too. Note, also, that the different implementations need to expose the same API so that it is possible to swap between them.

Reply:
I updated this accordingly, let me know if this is what you meant.